test/xtratest_flow_timeout.py

```diff
@@ -132,3 +132,59 @@ def test_external_lb_relay_timeout(prepare_ipv4, grpc_client, fast_flow_timeout)
 
 	grpc_client.dellbtarget(lb_name, neigh_ul_ipv6)
 	grpc_client.dellb(lb_name)
+
+def test_vip_nat_to_lb_on_another_vni(prepare_ipv4, grpc_client, port_redundancy):
+
+	if port_redundancy:
+		pytest.skip("Port redundancy is not supported for LB(vni1) <-> VIP/NAT(vni2) test")
+
+	lb_ul_ipv6 = grpc_client.createlb(lb_name, vni1, lb_ip, "tcp/80")
+	lb_vm1_ul_ipv6 = grpc_client.addlbprefix(VM1.name, lb_pfx)
+	grpc_client.addlbtarget(lb_name, lb_vm1_ul_ipv6)
+	lb_vm2_ul_ipv6 = grpc_client.addlbprefix(VM2.name, lb_pfx)
+	grpc_client.addlbtarget(lb_name, lb_vm2_ul_ipv6)
+
+	vip_ipv6 = grpc_client.addvip(VM3.name, vip_vip)
+	grpc_client.addfwallrule(VM2.name, "fw0-vm2", proto="tcp", dst_port_min=80, dst_port_max=80)
+	grpc_client.addfwallrule(VM1.name, "fw0-vm1", proto="tcp", dst_port_min=80, dst_port_max=80)
+
+	# Also test basic maglev behaviour
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM1.tap, 1242)
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM1.tap, 1242)
+	age_out_flows()
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM1.tap, 1242)
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM2.tap, 1234)
+
+	# Add/Remove dummy lb targets and the test the traffic againg
+	grpc_client.addlbtarget(lb_name, "cafe:dede::1")
+	grpc_client.addlbtarget(lb_name, "cafe:dede::ab")
+	grpc_client.addlbtarget(lb_name, "cafe:dede::de")
+	grpc_client.addlbtarget(lb_name, "cafe:dede::ff")
+
+	grpc_client.dellbtarget(lb_name, "cafe:dede::1")
+	grpc_client.dellbtarget(lb_name, "cafe:dede::ab")
+	grpc_client.dellbtarget(lb_name, "cafe:dede::de")
+	grpc_client.dellbtarget(lb_name, "cafe:dede::ff")
+
+	age_out_flows()
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM1.tap, 1242)
+	communicate_vip_lb(VM3, lb_ul_ipv6, vip_ipv6, vip_vip, VM2.tap, 1234)
+
+	grpc_client.delvip(VM3.name)
+
+	# NAT should behave the same, just test once (watch out for round-robin from before)
+	nat_ipv6 = grpc_client.addnat(VM3.name, nat_vip, nat_local_min_port, nat_local_max_port)
+	communicate_vip_lb(VM3, lb_ul_ipv6, nat_ipv6, nat_vip, VM2.tap, 1233)
+	grpc_client.delnat(VM3.name)
+
+	grpc_client.dellbtarget(lb_name, lb_vm2_ul_ipv6)
+	grpc_client.dellbprefix(VM2.name, lb_pfx)
+	grpc_client.dellbtarget(lb_name, lb_vm1_ul_ipv6)
+	grpc_client.dellbprefix(VM1.name, lb_pfx)
+	grpc_client.dellb(lb_name)
+
+	grpc_client.delfwallrule(VM2.name, "fw0-vm2")
+	grpc_client.delfwallrule(VM1.name, "fw0-vm1")
+
+	# NOTE: this test, just like in test_pf_to_vf.py
+	# cannot be run twice in a row, since the flows need to age-out
```

Comment:
I get why you needed to do the maglev test in this test file, but I think the original LB-VIP test should still be present in the standard test file too, as the `xtratest_flow_timeout.py` should only test the special cases that use timeout.

I would suggest removing the NAT part here and using the old test code without roundroubin/malgev part in the original location.

Reply:
We can not run the original LB-VIP test as with the new maglev implementation and random generated underlay ipv6 addresses, the backend selection is not deterministic.
I am running this here as this one is built with disabled random underlay ipv6 generation.

Reply:
That's what I am saying, that the maglev test needs to stay as-is here. The original location should only test whether communication NAT->LB in differing VNIs works.

Meaning the original test would not contain any add/remove LB targets, no flow aging, and you would only use one target, thus no randomness is present.